mpt/impl/smtp/core/src/main/java/org/apache/james/mpt/smtp/jmap/JMAPVacationTestSmtpTest.java

```diff
@@ -0,0 +1,137 @@
+import static com.jayway.restassured.config.EncoderConfig.encoderConfig;
+import static com.jayway.restassured.config.RestAssuredConfig.newConfig;
+import static org.hamcrest.Matchers.equalTo;
+
+import java.net.InetAddress;
+import java.util.Locale;
+
+import javax.inject.Inject;
+
+import org.apache.james.jmap.JmapAuthentication;
+import org.apache.james.jmap.api.access.AccessToken;
+import org.apache.james.mpt.script.AbstractSimpleScriptedTestProtocol;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.TemporaryFolder;
+import org.testcontainers.containers.GenericContainer;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.ImmutableList;
+import com.google.common.net.InetAddresses;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.http.ContentType;
+
+public class JMAPVacationTestSmtpTest extends AbstractSimpleScriptedTestProtocol {
+
+    public static final String USER = "bob";
+    public static final String DOMAIN = "mydomain.tld";
+    public static final String USER_AT_DOMAIN = USER + "@" + DOMAIN;
+    public static final String PASSWORD = "secret";
+    public static final String HTTP_LOCALHOST = "http://127.0.0.1";
+    public static final String YOPMAIL_COM = "yopmail.com";
+    public static final String MYDOMAIN_TLD = "mydomain.tld";
+    public static final String LOCALHOST = "127.0.0.1";
+    public static final String REASON = "Message explaining my wonderful vacations";
+
+    private final TemporaryFolder folder = new TemporaryFolder();
+    private final GenericContainer fakeSmtp = new GenericContainer("weave/rest-smtp-sink:latest");
+
+    @Rule
+    public final RuleChain chain = RuleChain.outerRule(folder).around(fakeSmtp);
+
+    @Inject
+    private static JmapSmtpHostSystem hostSystem;
+    private int restSMTPPort;
+    private String restSMTPBasedUPI;
+
+    public JMAPVacationTestSmtpTest() throws Exception {
+        super(hostSystem, USER_AT_DOMAIN, PASSWORD, "/org/apache/james/smtp/scripts/");
+    }
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        InetAddress containerIp = InetAddresses.forString(fakeSmtp.getContainerInfo().getNetworkSettings().getIpAddress());
+        hostSystem.getInMemoryDnsService()
+            .registerRecord(YOPMAIL_COM, new InetAddress[]{containerIp}, ImmutableList.of(YOPMAIL_COM), ImmutableList.of());
+        hostSystem.getInMemoryDnsService()
+            .registerRecord(DOMAIN, new InetAddress[]{InetAddresses.forString(LOCALHOST)}, ImmutableList.of(MYDOMAIN_TLD), ImmutableList.of());
+
+        restSMTPPort = 80;
+        restSMTPBasedUPI = "http://" + containerIp.getHostAddress();
+        RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
+    }
+
+    @Test
+    public void jmapVacationShouldWork() throws Exception {
```

Comment:
I understand the role of this test, but why MPT ? You can simply send a mail with JMAP I guess.